app/mailers/generic_mailer.rb

```diff
@@ -46,17 +46,46 @@ def self.deliver(method, options = {})
     msg
   end
 
-  def self.deliver_queue(method, options = {})
+  def self.deliver_queue(method, options = {}, queue_options = {})
     return unless MiqRegion.my_region.role_assigned?('notifier')
 
     _log.info("starting: method: #{method} args: #{options} ")
     options[:attachment] &&= attachment_to_blob(options[:attachment])
-    MiqQueue.submit_job(
+
+    queue_options.reverse_merge!(
       :service     => "notifier",
       :class_name  => name,
       :method_name => 'deliver',
       :args        => [method, options]
     )
+
+    MiqQueue.submit_job(**queue_options)
+  end
+
+  def self.deliver_task(method, options = {})
+    msg = "Queued the action: [#{method}]"
+
+    task = MiqTask.create!(:state => MiqTask::STATE_QUEUED, :status => MiqTask::STATUS_OK, :message => msg)
+
+    # Fail the task if there is no servier with the notifier role in this region
+    unless MiqRegion.my_region.role_assigned?('notifier')
+      task.error(_("No server with notifier role in region"))
+      return task
+    end
+
+    queue_options = {
+      :miq_task_id  => task.id,
+      :miq_callback => {
```

Comment:
Not sure if we should use this here, but there's a method to set up a generic callback: `MiqTask.generic_action_with_callback`.

Reply:
Yeah I started using that, but it would have ended up duplicating most of the `deliver_queue` code where I wanted the _task method to wrap the _queue method.